File: frontend/src/app/picture/picture.service.ts
import {Injectable} from '@angular/core';
import {LogService} from '../log/log.service';
import 'rxjs/add/operator/toPromise';
import {Http, RequestOptions, Headers, Response} from "@angular/http";
import {Picture} from "../model/picture";
import {AuthenticationService} from "../authentication/authentication.service";
import {Observable} from "rxjs";
import {User} from "../model/user";
import {AuthorityComponent} from "../authority/authority.component";

@Injectable()
export class PictureService {

    constructor(private logService: LogService, private http: Http, private authenticationService: AuthenticationService) { }

    findAll(): Promise<Picture[]> {
        return this.http.get('picture/pictures')
            .toPromise()
            .then(response => response.json())
            .catch(error => {
                this.logService.write('Request failed: ' + error);
            });
    }

    findById(inputId: number): Promise<Picture> {
        return this.http.get('picture/pictures/:'+inputId)
            .toPromise()
            .then(response => response.json())
            .catch(error => {
                this.logService.write('Request failed: ' + error);
            });
    }

    findByName(inputText: string): Promise<Picture> {
        return this.http.get('picture/pictures/'+inputText)
            .toPromise()
            .then(response => response.json())
            .catch(error => {
                this.logService.write('Request failed: ' + error);
            });
    }

    delete(picture: Picture): Observable<boolean> {
        const body = JSON.stringify({id: picture.id});
        let authUser: User = AuthorityComponent.getCurrentUser();
        let options = new RequestOptions({
            headers: new Headers({
                'Content-Type': 'application/json',
                'x-auth-token': authUser.token
            })
        });
        return this.http.post('picture/delete', body, options)
            .map((response: Response) => {
                if (response.ok) {
                    return true;
                }
                return false;
            })
            .catch(this.handleServerError);
    }

    private handleServerError(error: Response) {
        return Observable.throw(error.json());
    }
}

File: src/main/java/by/intexsoft/picturegallery/service/PictureService.java
package by.intexsoft.picturegallery.service;

import by.intexsoft.picturegallery.model.Picture;

import java.util.List;

/**
 * Interface for {@link Picture} service
 */
public interface PictureService {

    /**
     * Method adds picture to database
     * @param picture
     * @return {@link Picture}
     */
    Picture add(Picture picture);

    /**
     * Method removes picture from database
     * @param id
     */
    void delete(int id);

    /**
     * Method search picture in database by id
     * @param id
     * @return {@link Picture}
     */
    Picture findById(int id);

    /**
     * Method search picture in database by name
     * @param name
     * @return {@link Picture}
     */
    Picture findByName(String name);

    /**
     * Method edit picture in database
     * @param user
     * @return {@link Picture}
     */
    Picture edit(Picture user);

    /**
     * Method get all pictures from database
     * @return List<{@link User}>
     */
    List<Picture> findAll();
}